js/src/builder/fixedsizelist.ts

```diff
@@ -24,10 +24,16 @@ export class FixedSizeListBuilder<T extends DataType = any, TNull = any> extends
     public setValue(index: number, value: T['TValue']) {
         const [child] = this.children;
         const start = index * this.stride;
-        for (let i = -1, n = value.length; ++i < n;) {
+        for (let i = -1, n = this.stride; ++i < n;) {
             child.set(start + i, value[i]);
         }
     }
```

Comment:
I'm probably missing where the nulls are handled. This seems like it would error if `value` was `null`.

Reply:
The contract is that `setValue()` is only [called](https://github.com/apache/arrow/blob/0de45b90255c69f45c2e9073dd21be8784ebc8e0/js/src/builder.ts#L246) if the value is not null.